src/components/blog/BlogPostTable.tsx

```diff
@@ -0,0 +1,274 @@
+  position: relative;
+  min-width: 100px;
+  height: 60px;
+  background-image: url(${props => props.src});
+  background-size: cover;
+  background-position: center;
+  display: flex;
+  border-radius: 4px;
+  align-items: center;
+  margin-right: 15px;
+`
+const StyledIcon = styled.div`
+  position: absolute;
+  top: 0.5rem;
+  right: 0.5rem;
+`
+
+const StyledTitle = styled.div`
+  color: var(--gray-darker);
+  line-height: normal;
+  letter-spacing: 0.2px;
+  cursor: pointer;
+  display: -webkit-box;
+  -webkit-box-orient: vertical;
+  -webkit-line-clamp: 5;
+  overflow: hidden;
+`
+
+const DetailItem = styled(Menu.Item)`
+  padding: 0.5rem 1rem;
+`
+
+const filterIcon = (filtered: boolean) => <SearchOutlined style={{ color: filtered ? 'var(--primary)' : undefined }} />
+
+type BlogPostListColumn = {
+  id: String
+  authorName?: string | null | undefined
+  coverUrl: string | null
+  videoUrl: string | null
+  title: string
+  views?: number | null
+  publishedAt: Date | null
+  pinned_at: string | null
+  roles?: {
+    name: string | null | undefined
+    memberId?: string | null | undefined
+  }[]
+  postMerchandises: { id: string | null | undefined }[]
+}
+
+const BlogPostTable: React.VFC<{ blogPostData: BlogPostListColumn[]; postTableType: String; onRefetch: () => void }> =
+  ({ blogPostData = [], onRefetch = () => {}, postTableType = '' }) => {
+    const { formatMessage } = useIntl()
+    const [searchTitle, setSearchTitle] = useState<string>('')
+    const [searchAuthor, setSearchAuthor] = useState<string>('')
+    const [detailLoading, setDetailLoading] = useState(false)
+    const [blogDisplayData, setBlogDisplayData] = useState<BlogPostListColumn[]>([])
+    const [updatePostPinat] = useMutation<hasura.UPDATE_POST_PINNED_AT, hasura.UPDATE_POST_PINNED_ATVariables>(
+      UPDATE_POST_PINNED_AT,
+    )
+
+    const blogPostPinatNumber = blogPostData.filter(post => !!post.pinned_at).length
+
+    const handleUpload = (id: String | null, updateTime: Date | null, status: String) => {
+      if (blogPostPinatNumber === 3 && status === 'update') {
+        return message.warning(formatMessage(blogMessages.text.uploadPinnedAtLimmited))
+      }
+      setDetailLoading(true)
+
+      updatePostPinat({
+        variables: {
+          postId: id,
+          pinnedAt: updateTime,
+        },
+      })
+        .then(() => {
+          message.success(formatMessage(commonMessages.event.successfullySaved))
+          onRefetch()
+        })
+        .catch(handleError)
+        .finally(() => setDetailLoading(false))
+    }
+
+    useEffect(() => {
+      let result
+      if (searchTitle === '' && searchAuthor === '') {
+        result = blogPostData
+      } else {
+        result = blogPostData.filter(
+          post => post.title?.includes(searchTitle) && post.authorName?.includes(searchAuthor),
+        )
+      }
+      setBlogDisplayData(result)
+    }, [searchTitle, searchAuthor, blogPostData])
```

Comment:
NITS: try useMemo, more readable `to me`

Is this useEffect really needed ?

`example`
```
const filteredBlogPost = searchTitle === '' && searchAuthor === '' ? blogPostData : blogPostData.filter(
          post => post.title?.includes(searchTitle) && post.authorName?.includes(searchAuthor)
        )
...
<Table>
           ...
            dataSource={filteredBlogPost}
          />
        
```
is this work too?

Reply:
fixed it